discovery-provider/src/tasks/entity_manager/developer_app.py

```diff
@@ -90,25 +114,56 @@ def validate_developer_app_tx(params: ManageEntityParameters, metadata):
             f"Invalid {params.action} Developer App Transaction, user does not match signer"
         )
     if params.action == Action.DELETE:
-        if metadata["address"] not in params.existing_records[EntityType.DEVELOPER_APP]:
+        if not address:
+            raise Exception(
+                f"Invalid {params.action} Developer App Transaction, address is required and was not provided"
+            )
+
+        if address not in params.existing_records[EntityType.DEVELOPER_APP]:
             raise Exception(
                 f"Invalid Delete Developer App Transaction, developer app with address {metadata['address']} does not exist"
             )
         existing_developer_app = params.existing_records[EntityType.DEVELOPER_APP][
-            metadata["address"]
+            address
         ]
         if user_id != existing_developer_app.user_id:
             raise Exception(
                 f"Invalid Delete Developer App Transaction, user id {user_id} does not match given developer app address"
             )
     elif params.action == Action.CREATE:
+        if not metadata["app_signature"]:
+            raise Exception(
+                "Invalid Create Developer App Transaction, app signature is required and was not provided"
+            )
+        if (
+            not isinstance(metadata["app_signature"], dict)
+            or not metadata["app_signature"]
+            .get("message", "")
+            .startswith("Creating Audius developer app at ")
+            or not is_within_6_hours(
+                (metadata["app_signature"].get("message", "").split())[-1]
+            )
+        ):
+            raise Exception(
+                "Invalid Create Developer App Transaction, app signature provided does not have correct message"
+            )
+        try:
+            address = get_app_address_from_signature(metadata["app_signature"])
```

Comment:
should this recovered address apply to both create / delete and should this assert it matches address?

Reply:
see above comment but lmk if it's whack